### PR TITLE
Try to fix some of the decay

### PR DESCRIPTION
**Pull request by Powersource**
You could almost say the project has been _rusting_ for two years. This
commit doesn't compile, but it didn't before either so I'm not feeling too bad
about pushing it. And of course just as I was finishing up on this
branch from the master branch I see that you've actually _changed_ stuff
in the backup branch. Did not expect that. But it doesn't look like
changes that affect what I did anyway so hopefully it can merge without
too many problems. This is probably where I'm leaving it too, this is
taking far too much time and I could spend this time learning rust like
a sane person would.

(I was about to ask you why you didn't use cargo but I just looked it up and I think it had barely started existing when you pushed the last commit)

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -4,8 +4,11 @@ extern crate native;
 extern crate rsfml;
 extern crate collections;
 
-#[feature(phase)] extern crate log;
-#[phase(syntax, link)]
+// idk wth these lines were used for, couldn't find any other references
+// and the compiler complained. If the code doesn't use it maybe the
+// compiler does?
```
Comment (Powersource) [inline src/main.rs:14]: I just want to add that these comments are just slightly off from my final conclusion (if you can call it that): that the program only benefited from their removal.  
https://stackoverflow.com/questions/37903254/troubles-deriving-show-when-updating-very-old-code

**Comment by Powersource** [comment]
The current error:

```
error[E0038]: the trait `sfml::graphics::Drawable` cannot be made into an object
  --> src/engine/rendering.rs:76:5
   |
76 |     renderers: VecDeque<&'a Drawable>,
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `sfml::graphics::Drawable` cannot be made into an object
   |
   = note: method `draw` has generic type parameters

error[E0038]: the trait `sfml::graphics::Drawable` cannot be made into an object
  --> src/engine/rendering.rs:88:5
   |
88 |       pub fn push(&mut self, renderer: &'a Drawable) {
   |  _____^ starting here...
89 | |         self.renderers.push_back(renderer);
90 | |     }
   | |_____^ ...ending here: the trait `sfml::graphics::Drawable` cannot be made into an object
   |
   = note: method `draw` has generic type parameters
```

This is leading me into, for me, completely uncharted territory in the documentation. Any idea how to fix this?  
@bollu 

Relevant error documentation https://doc.rust-lang.org/error-index.html#method-has-generic-type-parameters  
I think it might be worth a shot to point the sfml dependency directly at their repo. The crates.io version is over a year old (it seems they're slowly working on a new release though) and they have [updated the trait declaration](https://github.com/jeremyletang/rust-sfml/commit/a02e08b00828fb66130987087afb708613ec7825) since then. Might try this tomorrow.

**Comment by bollu** [comment]
I see, let me check this out. For now, I'm going to merge this :)

**Comment by Powersource** [comment]
Ok I went ahead and merged instead :)

**Pull request merged**